common/frame.go

```diff
@@ -719,25 +712,42 @@ func (f *Frame) Evaluate(pageFunc goja.Value, args ...goja.Value) interface{} {
 	f.log.Debugf("Frame:Evaluate", "fid:%s furl:%q", f.ID(), f.URL())
 
 	rt := k6common.GetRuntime(f.ctx)
-	f.waitForExecutionContext("main")
-	i, err := f.mainExecutionContext.Evaluate(f.ctx, pageFunc, args...)
+
+	f.waitForExecutionContext(mainWorld)
+
+	opts := evaluateOptions{
+		forceCallable: true,
+		returnByValue: true,
+	}
+	result, err := f.evaluate(f.ctx, mainWorld, opts, pageFunc, args...)
 	if err != nil {
 		k6common.Throw(rt, err)
 	}
+
 	applySlowMo(f.ctx)
-	return i
+
+	return result
 }
 
 // EvaluateHandle will evaluate provided page function within an execution context
-func (f *Frame) EvaluateHandle(pageFunc goja.Value, args ...goja.Value) api.JSHandle {
+func (f *Frame) EvaluateHandle(pageFunc goja.Value, args ...goja.Value) (handle api.JSHandle) {
 	f.log.Debugf("Frame:EvaluateHandle", "fid:%s furl:%q", f.ID(), f.URL())
 
 	rt := k6common.GetRuntime(f.ctx)
-	f.waitForExecutionContext("main")
-	handle, err := f.mainExecutionContext.EvaluateHandle(f.ctx, pageFunc, args...)
+
+	f.waitForExecutionContext(mainWorld)
+
+	var err error
+	f.executionContextMu.RLock()
+	{
+		ec := f.executionContexts[mainWorld]
+		handle, err = ec.EvaluateHandle(f.ctx, pageFunc, args...)
```

Comment:
Here too we could use a nil check.

Since this is commonly used, let's add a `Frame.getContext(world)` method, and `Frame.hasContext(world)` could call it.

Reply:
Thanks. Yeah, but we'll still need to check for nil and type the same errors. I'm not sure what benefit `getContext` will provide us.

Reply:
> I'm not sure what benefit `getContext` will provide us.

Encapsulate the usage of `f.executionContextMu`. But yeah, we'd still need to check for nil.